src/Compilers/CSharp/Portable/Symbols/Source/SourceComplexParameterSymbol.cs

```diff
@@ -262,6 +262,36 @@ private ConstantValue DefaultSyntaxValue
                         }
                     }
 
+                    // Ensure availability of `DecimalConstantAttribute`.
+                    // For lambdas, check also explicit attribute application
+                    // because it will be re-synthesized in the delegate type.
+                    var isLambda = this is LambdaParameterSymbol;
+                    var defaultValueLocation = parameterEqualsValue?.Value.Syntax.Location ?? CSharpSyntaxNode?.Location ?? Locations.FirstOrDefault();
+                    if (isLambda
+                        ? _lazyDefaultSyntaxValue is { IsDecimal: true } ^
+                            (HasOptionalAttribute && DefaultValueFromAttributes is { IsDecimal: true })
```

Comment:
>HasOptionalAttribute [](http://example.com/codeflow?start=29&length=20)

Why is this important?

Reply:
Because parameter without `Optional` won't get synthesized delegate, i.e., `([DecimalConstant(1, 0, 0u, 0u, 11u)] decimal d) => { }` will have type `System.Action<decimal>`.

Reply:
After moving code around, this part is no longer present.